### PR TITLE
docs() Execution context suggested updates

### PR DESCRIPTION
**Pull request by johnbiundo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: content/exception-filters.md [hunk 1]
```diff
@@ -183,9 +183,9 @@ The `@Catch(HttpException)` decorator binds the required metadata to the excepti
...
 
-\*The reason for this level of abstraction is that `ArgumentsHost` functions in all contexts (e.g., the HTTP server context we're working with now, but also Microservices and Sockets). Later, we'll see how we can access the appropriate underlying arguments for **any** execution context with the power of `ArgumentsHost` and its helper functions. This will allow us to write generic exception filters that operate across all contexts.
+\*The reason for this level of abstraction is that `ArgumentsHost` functions in all contexts (e.g., the HTTP server context we're working with now, but also Microservices and Sockets). In the execution context chapter we'll see how we can <a href="https://docs.nestjs.com/fundamentals/execution-context#hopst-methods">access the appropriate underlying arguments</a> for **any** execution context with the power of `ArgumentsHost` and its helper functions. This will allow us to write generic exception filters that operate across all contexts.
```
Comment (kamilmysliwiec) [inline content/exception-filters.md]: > execution-context#hopst-methods

Invalid URL(?)

File: content/fundamentals/execution-context.md [hunk 2]
```diff
@@ -40,30 +57,46 @@ switchToHttp(): HttpArgumentsHost;
...
+  /**
+   * Returns the data object.
+   */
+  getData<T = any>(): T;
+  /**
+   * Returns the client object.
+   */
+  getClient<T = any>(): T;
+}
+```
```
Comment (kamilmysliwiec) [inline content/fundamentals/execution-context.md]: Can we add a single sentence between these two examples? (it looks slightly better 😄 )

File: content/fundamentals/execution-context.md [hunk 3]
```diff
@@ -79,18 +112,18 @@ export interface ExecutionContext extends ArgumentsHost {
...
 const className = ctx.getClass().name; // "CatsController"
 ```
 
-Having an ability to access references to both class and method opens many doors for us. Most importantly, it gives us an opportunity to read the metadata set through the `@SetMetadata()` decorator from within guards or interceptors.
+The ability to access references to both the current class and handler method provides great flexibility. Most importantly, it gives us the opportunity to access the metadata set through the `@SetMetadata()` decorator from within guards or interceptors. We cover this use case below.
 
 #### Reflection and metadata
 
-Nest provides the ability to attach custom **metadata** to route handlers through the `@SetMetadata()` decorator. We can then access this metadata from within our class to make certain decisions.
+Nest provides the ability to attach **custom metadata** to route handlers through the `@SetMetadata()` decorator (read more about custom metadata [here](https://docs.nestjs.com/guards#reflection)). We can then access this metadata from within our class to make certain decisions.
```
Comment (kamilmysliwiec) [inline content/fundamentals/execution-context.md]: > (read more about custom metadata [here](https://docs.nestjs.com/guards#reflection)).

Actually, I think this chapter will provide somewhat more information about custom metadata than guards(?)

File: content/fundamentals/execution-context.md [hunk 4]
```diff
@@ -191,9 +224,9 @@ const roles = this.reflector.get<string[]>('roles', context.getClass());
 const roles = this.reflector.get('roles', context.getClass());
 ```
 
-In addition, `Reflector` provides two other utility methods used to extract controller and method metadata at once.
+The `Reflector` class provides two other utility methods used to extract **both** controller and method metadata at once. This is helpful when, for example, default roles are set at the controller level, but overridden at the method (route) level.
```
Comment (kamilmysliwiec) [inline content/fundamentals/execution-context.md]: See DM (not sure if `overridden` is the best word here)

**Comment by johnbiundo** [comment]
@kamilmysliwiec comments resolved, a few minor improvements made (in Reflection and metadata section).  Ready for review.

**Comment by kamilmysliwiec** [comment]
Thank you :)

**Pull request merged**